napari/_qt/_tests/test_qt_provide_theme.py

```diff
@@ -32,13 +33,37 @@ def napari_experimental_provide_theme():
     reg = napari_plugin_manager._theme_data["TestPlugin"]
     assert isinstance(reg["dark-test-2"], Theme)
 
+    # set the viewer theme
     viewer.theme = "dark-test-2"
 
     # triggered when theme was added
     mock_add_theme.assert_called()
     mock_remove_theme.assert_not_called()
 
     # now, lets unregister the theme
+    # We didn't set the setting, so there should be no warning
+    with pytest.warns(None):
```

Comment:
if there will bo no wart, then why line 45. Can it be removed?

Reply:
I broke the tests up. 
So the first test doesn't set the new theme as a setting, just a viewer.theme
So I want to assert that no warning is generated when the theme is unregistered.
I've fixed the test based on pytest docs:
https://docs.pytest.org/en/7.0.x/how-to/capture-warnings.html#additional-use-cases-of-warnings-in-tests